app/assets/javascripts/discourse/components/topic-list-item.js.es6

```diff
@@ -29,175 +29,177 @@ export function navigateToTopic(topic, href) {
+      return true;
+    }
 
-      return false;
-    },
+    if (this.get("expandAllPinned")) {
+      return true;
+    }
 
-    showEntrance,
+    return false;
+  },
 
-    click(e) {
-      const result = this.showEntrance(e);
-      if (result === false) {
-        return result;
-      }
+  showEntrance,
 
-      const topic = this.get("topic");
-      const target = $(e.target);
-      if (target.hasClass("bulk-select")) {
-        const selected = this.get("selected");
+  click(e) {
+    const result = this.showEntrance(e);
+    if (result === false) {
+      return result;
+    }
 
-        if (target.is(":checked")) {
-          selected.addObject(topic);
-        } else {
-          selected.removeObject(topic);
-        }
-      }
+    const topic = this.get("topic");
+    const target = $(e.target);
+    if (target.hasClass("bulk-select")) {
+      const selected = this.get("selected");
 
-      if (target.hasClass("raw-topic-link")) {
-        if (wantsNewWindow(e)) {
-          return true;
-        }
-        return this.navigateToTopic(topic, target.attr("href"));
+      if (target.is(":checked")) {
+        selected.addObject(topic);
+      } else {
+        selected.removeObject(topic);
       }
+    }
 
-      if (target.closest("a.topic-status").length === 1) {
-        this.get("topic").togglePinnedForUser();
-        return false;
+    if (target.hasClass("raw-topic-link")) {
+      if (wantsNewWindow(e)) {
+        return true;
       }
+      return this.navigateToTopic(topic, target.attr("href"));
+    }
 
-      return this.unhandledRowClick(e, topic);
-    },
-
-    navigateToTopic,
+    if (target.closest("a.topic-status").length === 1) {
+      this.get("topic").togglePinnedForUser();
+      return false;
+    }
 
-    // Can be overwritten by plugins to handle clicks on other parts of the row
-    unhandledRowClick() {},
+    return this.unhandledRowClick(e, topic);
+  },
+
+  navigateToTopic,
+
+  highlight(opts = { isLastViewedTopic: false }) {
+    const $topic = this.$();
+    $topic
+      .addClass("highlighted")
+      .attr("data-islastviewedtopic", opts.isLastViewedTopic);
+
+    $topic.on("animationend", () => $topic.removeClass("highlighted"));
+  },
+
+  _highlightIfNeeded: function() {
+    // highlight the last topic viewed
+    if (this.session.get("lastTopicIdViewed") === this.get("topic.id")) {
+      this.session.set("lastTopicIdViewed", null);
+      this.highlight({ isLastViewedTopic: true });
+    } else if (this.get("topic.highlight")) {
+      // highlight new topics that have been loaded from the server or the one we just created
+      this.set("topic.highlight", false);
+      this.highlight();
+    }
+  }.on("didInsertElement")
+};
 
-    highlight(opts = { isLastViewedTopic: false }) {
-      const $topic = this.$();
-      $topic
-        .addClass("highlighted")
-        .attr("data-islastviewedtopic", opts.isLastViewedTopic);
+export default Ember.Component.extend(
+  bufferedRender(
```

Comment:
I don't think `$.extend` should be necessary here. You can include multiple mixins in an ember class:

`export default Ember.Component.extend(ListItemDefaults, bufferedRender( ... buffered specific methods ... ), { ... extra instance methods ... })`